Add reorder topologies command (#3296)
GeotrekCE/Geotrek-admin: docs/install/import.rst

      --traceback           Raise on CommandError exceptions
      --no-color            Don't colorize the command output.
      --force-color         Force colorization of the command output.
      --skip-checks         Skip system checks.

Import Signage
--------------

``sudo geotrek help loadsignage``

::

    usage: manage.py loadsignage [-h] [--use-structure] [--encoding ENCODING] [--name-field NAME_FIELD] [--type-field TYPE_FIELD] [--condition-field CONDITION_FIELD]
                                 [--structure-field STRUCTURE_FIELD] [--description-field DESCRIPTION_FIELD] [--year-field YEAR_FIELD] [--code-field CODE_FIELD]
                                 [--type-default TYPE_DEFAULT] [--name-default NAME_DEFAULT] [--condition-default CONDITION_DEFAULT] [--structure-default STRUCTURE_DEFAULT]
                                 [--description-default DESCRIPTION_DEFAULT] [--eid-field EID_FIELD] [--year-default YEAR_DEFAULT] [--code-default CODE_DEFAULT] [--version]
                                 [-v {0,1,2,3}] [--settings SETTINGS] [--pythonpath PYTHONPATH] [--traceback] [--no-color] [--force-color] [--skip-checks]
                                 point_layer

>     Load a layer with point geometries in te structure model

Reviewer: ```suggestion
    Load a layer with point geometries in the signage model
```

      -v {0,1,2,3}, --verbosity {0,1,2,3}
                            Verbosity level; 0=minimal output, 1=normal output, 2=verbose output, 3=very verbose output
      --settings SETTINGS   The Python path to a settings module, e.g. "myproject.settings.main". If this isn't provided, the DJANGO_SETTINGS_MODULE environment variable will be used.
      --pythonpath PYTHONPATH
                            A directory to add to the Python path, e.g. "/home/djangoprojects/myproject".
      --traceback           Raise on CommandError exceptions
      --no-color            Don't colorize the command output.
      --force-color         Force colorization of the command output.
      --skip-checks         Skip system checks.

Exploitation commands
=====================

Delete attachment from disk
---------------------------

When an attachment (eg. pictures) is removed, its file is not automatically removed from disk.
You have to run ``sudo geotrek clean_attachments`` manually or in a cron to remove old files.
After that, you should run ``sudo geotrek thumbnail_cleanup`` to remove old thumbnails.

Remove duplicate paths
----------------------

> It can happens during additions of paths with commands or directly in the application to have duplicate paths.

Reviewer: ```suggestion
Duplicate paths can appear while adding paths with commands or directly in the application.
```

Duplicate paths can cause some problems of routing for topologies, it can generates topologies representing multilinestring.

You have to run ``sudo geotrek remove_duplicate_paths``

> During the process of the command every topologies on a duplicate path will be set on the original path and the duplicate path deleted.

Reviewer: ```suggestion
During the process of the command, every topology on a duplicate path will be set on the original path, and the duplicate path will be deleted.
```

Unset structure on categories
-----------------------------

> If you need to remove some similar categories with different structures you may need to remove the structure of all categories

Reviewer: Pas très clair
```suggestion
Use this command if you wish to undo linking categories to structures for some models
```

    Unset structure in lists of choices and group choices with the same name.

    positional arguments:
      model                 List of choices to manage

    optional arguments:
      -h, --help            show this help message and exit
      --all                 Manage all models
      --list                Show available models to manage
      --version             show program's version number and exit
      -v {0,1,2,3}, --verbosity {0,1,2,3}
                            Verbosity level; 0=minimal output, 1=normal output, 2=verbose output, 3=very verbose output
      --settings SETTINGS   The Python path to a settings module, e.g. "myproject.settings.main". If this isn't provided, the DJANGO_SETTINGS_MODULE environment variable will be used.
      --pythonpath PYTHONPATH
                            A directory to add to the Python path, e.g. "/home/djangoprojects/myproject".
      --traceback           Raise on CommandError exceptions
      --no-color            Don't colorize the command output.
      --force-color         Force colorization of the command output.
      --skip-checks         Skip system checks.

.. danger::

>     You can't chose for each choice which set of category you want to unset structures. Every categories of each choice will

Reviewer: ```suggestion
    You can't chose for each choice which set of category you want to unset structures, it will happen for all categories
```

> Firstly, it creates (or finds) the category without structure associated.

Reviewer: ```suggestion
Firstly, if a categroy is linked to a structure, it creates the same category but with no structure associated.
```

Secondly, every element with this category get this new category.
Finally all old categories are removed.

Reorder topologies
------------------

> Every topologies have informations of which path they pass on and in which order.

Reviewer: ```suggestion
All topologies have information about which path they go through on and in which order.
```

> Actually, when an path is split in 2 by an other path, database add a new path.

Reviewer: ```suggestion
Actually, when a path is split in 2 by another path, a new path is added to the database.
```

> We need to add informations for every topologies which need to pass on this new path.

Reviewer: ```suggestion
We need to add information for all topologies that need to go through this new path.
```

This is badly managed at the moment, especially for the order of passage of the paths.
``sudo geotrek reorder_topologies``

> It removes a lot of useless information which can accelerate the process of edition of topologies afterward.

Reviewer: ```suggestion
It removes a lot of useless information which can accelerate the process of editing topologies afterward.
```

> During the process of this command, it tries to find a good order of passage on the paths which create

Reviewer: ```suggestion
During the process of this command, it tries to find a good order of passage on the paths which creates
```

> only one Linestring from the start to the end. It uses the wrong orders as much as possible. This command use the same algorithm to generate one Linestring

Reviewer: ```suggestion
only one Linestring from start to end. It stays as close as possible to the corrupted order. This command uses the same algorithm to generate one Linestring
```

when the order is not well managed during topologies' display.

.. danger::
    It can happens that this algorithm can't find any solution and will genereate a MultiLineString.
    This will be display at the end of the reorder

Automatication commands
-----------------------

> You can set up automatic commands by creating a file ``/etc/cron.d/geotrek_command`` that contains:

Reviewer: ```suggestion
You can set up automatic commands by creating a `cron` file under ``/etc/cron.d/geotrek_command`` that contains:
```
